controlplane/kubeadm/internal/etcd_client_generator.go

```diff
@@ -38,6 +38,8 @@ type EtcdClientGenerator struct {
 
 type clientCreator func(ctx context.Context, endpoints []string) (*etcd.Client, error)
 
+var errEtcdNodeConnection = errors.New("failed to connect to etcd node")
```

Comment:
Q: Should this be an error type so we can store the inner error vs aggregating two errors down below
(It probably requires to implement Unwrap() and Cause() to get it right)

Reply:
I used an error type along with `errors.As` initially (see commit ad0826a80997ab870469f800993fd110946d29a0), though @vincepri was more in favor of putting two errors into `Aggregate` and invoking `errors.Is` on it. I slightly prefer the `errors.As` approach as it seems somewhat clearer to me, but happy to follow any majority.

Reply:
If we encapsulate an error with another without implementing wrapping/unwrapping `errors.Is` or `errors.As` won't work anymore. The error aggregation I suggested above already implements these facilities and we're using it everywhere in the codebase